pkg/engine/journal.go

```diff
@@ -119,8 +119,10 @@ func (entries JournalEntries) Snap(base *deploy.Snapshot) *deploy.Snapshot {
 
 	manifest := deploy.Manifest{}
 	manifest.Magic = manifest.NewMagic()
-	return deploy.NewSnapshot(manifest, secretsManager, resources, operations)
 
+	snap := deploy.NewSnapshot(manifest, secretsManager, resources, operations)
+	err := snap.NormalizeURNReferences()
+	return snap, err
```

Comment:
Worth adding a call to validate here as well?

Reply:
Done.